prisma-filter/src/filter.parser.ts

```diff
@@ -104,9 +108,17 @@ export class FilterParser<TDto, TWhereInput> {
       ) {
         throw new Error(`Array filter value must be an Array<string|number|boolean>`);
       }
-      if (type === FilterOperationType.InStrings || type === FilterOperationType.NotInStrings)
+      if (
+        type === FilterOperationType.InStrings ||
+        type === FilterOperationType.NotInStrings ||
+        type === FilterOperationType.ArrayContains
+      )
         return rawValue;
-      if (type !== FilterOperationType.In && type !== FilterOperationType.NotIn) {
+      if (
+        type !== FilterOperationType.In &&
+        type !== FilterOperationType.NotIn //&&
+        // type !== FilterOperationType.ArrayContains
+      ) {
```

Comment:
Since the input values here are always strings (and Prisma is usually very strict with their typings, and you can't just pass a string for a number array), I think the best way is to offer two filter operations: `ArrayContains` and `ArrayContainsStrings`. For `ArrayContainsStrings`, you can directly return the raw value - and `ArrayContains` then correctly casts the values to numbers (you only have to include your commented-out code again, and it should work).

boolean arrays are currently not truly supported - but that's a library problem that I can fix myself if you don't want to add a bool conversion here.